main/2023/CG-10.md

```diff
@@ -113,4 +113,1850 @@ Use this space to add any agenda items that need to be scheduled, and scheduling
+
+JM: Architecture is very real, would be nice to know how good this is for the fullness of time, is this a feature that is desired? Are new languages coming up with newer concurrency model? Another question - does this pessimize wasm applications that donb’t use threads? Does this necessitate the design of a garbage collector that performs worse on single-threaded applications?
+
+HP: I can say something brief, the current implementation isn’t penalizing the main thread, we’re happy to share our story - we’re not sure if we’re at the good place yet, but we have projects in place that are working through it
+
+AR: I expect this direction to be inevitable. The question is now how hard we want to push forward on this. The next question is who would want to pick this up. As a proposal it’s relatively simple. You’ll have to a number of discussions with engine vendors.
+
+TL: Everything we talked about yesterday, we would also solve for this effort - salami slicing - how do we uncouple stuff? 
+
+AR: We should do it a part of the threading proposal 
+
+CW: We could legitimately call this share everything proposal
+
+PC: Should there be a profile? 
+
+AR: You’ll have a no-GC, no-threads profile - if you’ve excluded one you’ve excluded the other. Or if you need one you might need the other too
+
+CW: Just so I totally understand, did you win anything from being able to import the literal GC types you can import today or is this more relevant for post-MVP extensions?
+
+AR: Gives you a way of paramerarizing over the type you’re operating on, to the module its only a reference type, so you’re not tied to what the thing is, the string implementation is a special type
+
+CW: But would you ever try to import a less precise struct type by someone else or is that too much granularity
+
+AR: Like a supertype? I don’t have a good use case for that. Maybe that exists in some systems but honestly I don’t know.
+
+CW: That’s all I was hoping to hear
+
+AR: Not sure if that adds any significant complexity. It’s just subtype checking and you can restrict the syntax on one side of that check or not, other than that I don’t think it has one real implication on implementations.
+
+FM: You can’t compile an offset to a structure.. Maybe too specific
+
+AR: What I meant is there is no implication for engines to allow this but disallowing that might be a nice example of excluding some things, knowing you have a common header for structs that define a different module without having to know the extent of all of them. That’s a good example. Might be worth just doing it, but Post-Post-MVP.
+
+<AR Presenting>
+
+TL: There’s one thorny but trivial problem, the type section comes before the import section
+
+AR: Yes, I should have that on the next slide. There is also one minor thing, we want to allow type definitions for that because someone might want to include i31 as their implementation for something.
```

Comment:
```suggestion
AR: Yes, I should have that on the next slide. There is also one minor thing, we want to allow type definitions with abstract heap types because someone might want to include i31 as their implementation for something.
```